src/modules/bulletproofs/bulletproofs_pp_norm_product_impl.h

```diff
@@ -139,4 +139,251 @@ static int secp256k1_bulletproofs_commit(
+    const secp256k1_bulletproofs_generators* g_vec,
+    const secp256k1_scalar* n_vec,
+    size_t n_vec_len,
+    const secp256k1_scalar* l_vec,
+    size_t l_vec_len,
+    const secp256k1_scalar* c_vec,
+    size_t c_vec_len,
+    const secp256k1_ge* commit
+) {
+    secp256k1_scalar q, r = *r_ch;
+    secp256k1_sha256 sha256;
+    unsigned char ser_commit[33];
+    size_t start_idx = 0;
+    ec_mult_x_cb_data x_cb_data;
+    ec_mult_r_cb_data r_cb_data;
+    size_t g_len = n_vec_len, h_len = l_vec_len;
+    const size_t G_GENS_LEN = g_len;
+    /* The initial pointers to the callback function remain the same. We mutate
+     * the values and len in loop*/
+    secp256k1_scalar *ns, *ls, *cs;
+    secp256k1_ge *gs, comm = *commit;
+    size_t scratch_checkpoint;
+    size_t log_n = secp256k1_bulletproofs_pp_log2(g_len), log_m = secp256k1_bulletproofs_pp_log2(h_len);
+    size_t n_rounds = log_n > log_m ? log_n : log_m;
+
+    /* Check proof sizes.*/
+    if (*proof_len < 65 * n_rounds + 64) {
+        return 0;
+    }
+
+    if (g_vec->n != (n_vec_len + l_vec_len) || l_vec_len != c_vec_len) {
+        return 0;
+    }
+
+    if (!secp256k1_check_power_of_two(n_vec_len) ||  !secp256k1_check_power_of_two(c_vec_len)) {
+        return 0;
+    }
+    /* We can get away with allocating only half the size of arrays and unrolling the first iteration of the loop.
+     This would increase the code complexity and can be done as an optimization in a future PR. */
+    scratch_checkpoint = secp256k1_scratch_checkpoint(&ctx->error_callback, scratch);
+    ns = (secp256k1_scalar*)secp256k1_scratch_alloc(&ctx->error_callback, scratch, g_len * sizeof(secp256k1_scalar));
+    ls = (secp256k1_scalar*)secp256k1_scratch_alloc(&ctx->error_callback, scratch, h_len * sizeof(secp256k1_scalar));
+    cs = (secp256k1_scalar*)secp256k1_scratch_alloc(&ctx->error_callback, scratch, h_len * sizeof(secp256k1_scalar));
+    gs = (secp256k1_ge*)secp256k1_scratch_alloc(&ctx->error_callback, scratch, (g_len + h_len) * sizeof(secp256k1_ge));
+    if (ns == NULL || ls == NULL || cs == NULL || gs == NULL) {
+        secp256k1_scratch_apply_checkpoint(&ctx->error_callback, scratch, scratch_checkpoint);
+        return 0;
+    }
+    memcpy(ns, n_vec, g_len * sizeof(secp256k1_scalar));
+    memcpy(ls, l_vec, h_len * sizeof(secp256k1_scalar));
+    memcpy(cs, c_vec, h_len * sizeof(secp256k1_scalar));
+    memcpy(gs, g_vec->gens, (g_len + h_len) * sizeof(secp256k1_ge));
+
+    x_cb_data.n = ns;
+    x_cb_data.g = gs;
+    x_cb_data.l = ls;
+    x_cb_data.G_GENS_LEN = G_GENS_LEN;
+
+    r_cb_data.n1 = ns;
+    r_cb_data.g1 = gs;
+    r_cb_data.l1 = ls;
+    r_cb_data.G_GENS_LEN = G_GENS_LEN;
+    secp256k1_scalar_sqr(&q, &r);
+
+
+    secp256k1_sha256_initialize(&sha256);
+    secp256k1_sha256_write(&sha256, transcript_hash32, 32);
+    secp256k1_fe_normalize_var(&comm.x);
+    secp256k1_fe_normalize_var(&comm.y);
+    ser_commit[0] = 0x02 | secp256k1_fe_is_odd(&comm.y);
```

Comment:
In a726d5e62ef188f0549ce2ba1a6b59d01f897255:

I'm curious why you take pains here to encode this as a SECG point, but don't use the existing point serialization methods. I have some old BP code that does stuff like this but only if:
* I'm not able to modify the point I'm serializing (but you can, since you're normalizing its coordinates in place)
* I need to do the serialization in constant time (but we don't need to, since this is a public point .... and in any case you're not, because you're calling `normalize_var`)

Reply:
Also Liam points out that this LOC is not present in your other branch `bpp_final`....is this PR out of date?

Reply:
Ah, this code is removed in aec736da964be3db5dd137b824ed0b4cb612db51 (commit 15 of #207). Ok.

Reply:
This PR now uses the new FS scheme.